### PR TITLE
Feature/19 create doors and levels game

### PR DESCRIPTION
**Pull request by vsatkov**
Console game - Create doors and levels

### CODE REVIEW

**Comment by vsatkov** [comment]
issue #19

**Comment by vsatkov** [comment]
> 
> 
> The program works and executes the logic, but it seems to me that we should do more small universal methods and eliminate duplication of code. You can also work on the design and interfaces.

Thank you for you comment. I will consider it in my next home task.

**Comment by chesdenis** [comment]
@Dasha529 , could you please review this pull request?

File: HW1/Valeriy_Satkov/CreateDoorsAndLevels/CreateDoorsAndLevels/Program.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+{
+    class Program
+    {
+        public void Start()
+        {
+            List<int[]> list = new List<int[]>();
+
+            Random random = new Random();
+            // list.Add(new int[] { 2, 4, 3, 1, 0 }); // simple test
+            list.Add(new int[] { random.Next(1, 10), random.Next(1, 10), random.Next(1, 10), random.Next(1, 10), 0 });
```
[the remark this answers is not in the document]
Comment (vsatkov) [reply HW1/Valeriy_Satkov/CreateDoorsAndLevels/CreateDoorsAndLevels/Program.cs]: This is a piece from the previous commit, but you're right. Now I use 'for' in this part.

File: HW1/Valeriy_Satkov/CreateDoorsAndLevels/CreateDoorsAndLevels/Program.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+                if (selectedNumber > 0)
+                {
+                    // Create "next level" numbers where numbers values calculate using formula: [number on previous level] * [choosed number on previous level].
+                    list.Add(new int[] {
+                    list[level][0] * selectedNumber,
+                    list[level][1] * selectedNumber,
+                    list[level][2] * selectedNumber,
+                    list[level][3] * selectedNumber,
+                    list[level][4] * selectedNumber });
```
[the remark this answers is not in the document]
Comment (vsatkov) [reply HW1/Valeriy_Satkov/CreateDoorsAndLevels/CreateDoorsAndLevels/Program.cs]: Thank you for noticing the unnecessary double-depth list. It much better to make new numbers everytime.
I measured difference between 'for' with StringBuilder and string-interpolation. Finally I haven't significant differences but 'for' will be more comfortable in future if we use more variable.

File: HW1/Valeriy_Satkov/CreateDoorsAndLevels/CreateDoorsAndLevels/Program.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                    Console.WriteLine($"We select number {selectedNumber} and go to next level: {list[level + 1][0]} {list[level + 1][1]} {list[level + 1][2]} {list[level + 1][3]} {list[level + 1][4]} ({list[level][0]}x{selectedNumber} {list[level][1]}x{selectedNumber} {list[level][2]}x{selectedNumber} {list[level][3]}x{selectedNumber} {list[level][4]}x{selectedNumber})");
+                }
+                else if (level > 0)
+                {
+                    list.RemoveAt(level); // remove level                    
+
+                    // "We select number 0 and go to previous level: 4 8 6 2 0"
+                    Console.WriteLine($"We select number 0 and go to previous level: {list[level - 1][0]} {list[level - 1][1]} {list[level - 1][2]} {list[level - 1][3]} {list[level - 1][4]}");
+                }
+            } while (!(selectedNumber == 0 && level == 0));
```
[the remark this answers is not in the document]
Comment (vsatkov) [reply HW1/Valeriy_Satkov/CreateDoorsAndLevels/CreateDoorsAndLevels/Program.cs]: Yeah, Denis talked us about it. I will correct it later. Thanks for noticing.

**Pull request merged**